libtiledbsoma/src/soma/soma_array.cc

```diff
@@ -1039,107 +1039,107 @@ void SOMAArray::_set_soma_joinid_shape_helper(
                 }
                 ndrect.set_range<int64_t>(dim_name, 0, newshape - 1);
                 continue;
+            }
```

Comment:
Optional: I think switching this to an if/else instead of the continue would help with clarity and preventing similar bugs in the future.